solutions/CF Packmen.mdx

```diff
@@ -0,0 +1,294 @@
+---
+id: cf-packmen
+source: CF
+title: Packmen
+author: Kevin Sheng
+---
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+Binary search on the minimum time it will take for all the Packmen to eat all the pellets.
+For each Packman, we can greedily calculate the maximum number of pellets that it can eat in the given amount of time and then remove those pellets.
+Note that a Packman will visit all pellets between the leftmost and rightmost pellets a it eats, so it can eat all of them without any time penalty. Also, they must be non-intersecting because if they were, either Packman can ignore the pellets of the other Packman.
```

Comment:
Add clarification sentence "This means that all packmen will eat contiguous segments," or "All packmen will eat a continuous range of pellets" or something along these lines.

Reply:
Maybe phrase it like this?
```
Suppose we want to know whether the Packmen can eat all the pellets in $d$ units of time.

First, notice a few observations:
 * Each packman will eat a continuous subarray of the pellets, because a packman will visit all pellets between the leftmost and rightmost ones, and thus can eat all pellets between them with no additional time penalty.
 * No two packmen need to eat overlapping subarrays, as the region of overlap can be assigned to one of the packman and be ignored by the other
 * The leftmost packman will eat the leftmost subarray of pellets, because if this were not the case, the leftmost packman will necessarily cross over another packman. If this occurs, simply simulate (prevent) the crossing by reversing the directions of both packmen instead. This ensures the leftmost packman is always to the left of all other packmen, and thus will eat a subarray of pellets to the left of all others

We can repeatedly greedily assign the leftmost packman to the largest possible subarray of pellets that includes the leftmost pellet. Using this strategy, we can check whether all pellets can be eaten when the packmen are limited to $d$ units of time.
```